### PR TITLE
1.20.1

### PR DESCRIPTION
**Pull request by Brokkonaut**
Update to 1.20.1

### CODE REVIEW

**Comment by TheRealJake12** [comment]
upload an actions build here
we need something for now

File: src/main/java/foundationgames/enhancedblockentities/EBESetup.java [hunk 1]
```diff
@@ -61,40 +75,67 @@ public static void setupRRPSigns() {
         ResourceUtil.addSignBlockStates("mangrove_sign", "mangrove_wall_sign", p);
         ResourceUtil.addSignBlockStates("crimson_sign", "crimson_wall_sign", p);
         ResourceUtil.addSignBlockStates("warped_sign", "warped_wall_sign", p);
+        ResourceUtil.addSignBlockStates("bamboo_sign", "bamboo_wall_sign", p);
```
Comment (MeeniMc) [inline src/main/java/foundationgames/enhancedblockentities/EBESetup.java:82]: Aren't the new cherry trees missing from these lists?

File: gradle.properties [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-fabric_version=0.77.0+1.19.4
+fabric_version=0.83.0+1.20.1
 
-mod_version = 0.8+1.19.4
+mod_version = 0.8.1+1.20.1
 maven_group = foundationgames
 archives_base_name = enhancedblockentities
 
-arrp_version=0.6.7
+arrp_version=0.6.8
```
Comment (MeeniMc) [inline gradle.properties]: Can't find that version of arrp where did you get it ?

File: src/main/java/foundationgames/enhancedblockentities/EBESetup.java (hunk 1 above)
Comment (Brokkonaut) [reply src/main/java/foundationgames/enhancedblockentities/EBESetup.java:82]: > Aren't the new cherry trees missing from these lists?

You are right, I will fix that

File: gradle.properties (hunk 2 above)
Comment (Brokkonaut) [reply gradle.properties]: > Can't find that version of arrp where did you get it ?

https://github.com/Brokkonaut/ARRP

(i hope there will be an official release soon, but for now it is that)

Comment (MeeniMc) [reply gradle.properties]: The mod compiles and run with arrp 0.6.7

**Review by MeeniMc — APPROVED** [review]

File: gradle.properties [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 modmenu_version=6.1.0-rc.4
```
Comment (MeeniMc) [inline gradle.properties]: modmenu version required 7.0.1

**Comment by TheRealJake12** [comment]
> upload an actions build here we need something for now

**Comment by Brokkonaut** [comment]
> > upload an actions build here we need something for now

Just clone the repo and use ```./gradlew build```

**Comment by TheRealJake12** [comment]
> > > upload an actions build here we need something for now
> 
> Just clone the repo and use `./gradlew build`

I'm not a java programmer but alr, I'll try

File: gradle.properties (hunk 2 above)
Comment (JanLukasGithub) [reply gradle.properties]: @Brokkonaut can you make a PR for that version bump? Maybe that'll make the creator of ARRP do something.

**Comment by duplexsystem** [comment]
I get missing texture particles when breaking a cherry sign

File: gradle.properties (hunk 2 above)
Comment (Brokkonaut) [reply gradle.properties]: > @Brokkonaut can you make a PR for that version bump? Maybe that'll make the creator of ARRP do something.

Is was not necessary, I reverted that change. This PR now compiles against ARRP 0.6.7

**Comment by Brokkonaut** [comment]
> I get missing texture particles when breaking a cherry sign

should be fixed now

**Comment by fxmorin** [comment]
Performance seems much worse when using this fork, not sure what the cause is. Didn't check

**Comment by Rektroth** [comment]
Should the Gradle wrapper be updated to 8.1.1? I get this error otherwise:

```
* What went wrong:
Could not open cp_settings generic class cache for settings file 'C:\Users\Rektroth\source\repos\EnhancedBlockEntities\settings.gradle' (C:\Users\Rektroth\.gradle\caches\8.0\scripts\9i8ii8b4tcz62c0ak7z8aquwy).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 64
```

Not sure why. I'm using Java 17, but it acts like I'm using 20.

**Comment by Guerdi** [comment]
unfortunately incompatible with LambdaBetterGrass, block entities become invisible

**Comment by ImGRUI** [comment]
> unfortunately incompatible with LambdaBetterGrass, block entities become invisible

Not fork issue (https://github.com/LambdAurora/LambdaBetterGrass/issues/83), but
You can fix that by changing 27 line in _src\main\java\foundationgames\enhancedblockentities\mixin\LifecycledResourceManagerImplMixin.java_, replace "INVOKE_ASSIGN" to  "INVOKE" ( just saw pull request https://github.com/FoundationGames/EnhancedBlockEntities/pull/147 with this and LambdAurora mentioned this fix in the LambdaBetterGrass issue)
If it doesn't work, change experimental features to another value. Works for me

**Comment by Guerdi** [comment]
I hadn't updated my modpack since 1.19 so I missed this problem, thank you very much for the quick response!
Unfortunately, now the problem is that the block destruction particles are missing (black and purple).
It's still less disturbing than before, I guess...

**Comment by ImGRUI** [comment]
> I hadn't updated my modpack since 1.19 so I missed this problem, thank you very much for the quick response! Unfortunately, now the problem is that the block destruction particles are missing (black and purple). It's still less disturbing than before, I guess...

Hmm, i don't have that problem. Turn off Experimental Features, maybe it help.

**Review by FoundationGames — COMMENTED** [review]
Update looks good, will merge eventually
Some accessor methods in `SignBlockEntityRenderAccessor` appear to be commented out, if there's a reason for it please let me know; if not, I would prefer them removed.

The final update to 1.20 may take some time even after merging this, since EBE will need to include block models and technical changes for the new decorated pots, which also use entity rendering.

**Comment by TheRealJake12** [comment]
> Update looks good, will merge eventually Some accessor methods in `SignBlockEntityRenderAccessor` appear to be commented out, if there's a reason for it please let me know; if not, I would prefer them removed.
> 
> The final update to 1.20 may take some time even after merging this, since EBE will need to include block models and technical changes for the new decorated pots, which also use entity rendering.

oh my god he's awake

**Comment by TiboOpGithub** [comment]
Is there a safe version we can already use?

**Pull request merged**